### DISCUSSION TITLE
Calling core.setFailed() doesn't mark step as failed

### DISCUSSION THREAD

**Discussion by williamhaw** [community/community#25165] [Actions]
<p>Even though I called core.setFailed() in my error handling code, it didn’t mark the step as failed. I would like to be able to give users the chance to retry the workflow. I’m not sure if I’m doing something wrong.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/d/d3323a847a8acba9922a76970f4b0c50fc6c453d.png" data-download-href="/uploads/short-url/u8kmlNwIhR0fQq69V9UhHUYWMu9.png?dl=1" title="setfailed_screenshot.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/160841/181089211-26375b7c-4e3f-4ae9-8577-84541f138764.png" alt="setfailed_screenshot.png" data-base62-sha1="u8kmlNwIhR0fQq69V9UhHUYWMu9" width="690" height="317" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/d/d3323a847a8acba9922a76970f4b0c50fc6c453d_2_690x317.png, https://aws1.discourse-cdn.com/github/optimized/2X/d/d3323a847a8acba9922a76970f4b0c50fc6c453d_2_1035x475.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/d/d3323a847a8acba9922a76970f4b0c50fc6c453d_2_1380x634.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/d/d3323a847a8acba9922a76970f4b0c50fc6c453d_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">setfailed_screenshot.png</span><span class="informations">1579×727 24.6 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>
<p>Issue summarized here: <a href="https://github.com/williamhaw/setup-wiremock-action/issues/2" rel="nofollow noopener">https://github.com/williamhaw/setup-wiremock-action/issues/2</a></p>
<p>Code: <a href="https://github.com/williamhaw/setup-wiremock-action/blob/master/index.js" rel="nofollow noopener">https://github.com/williamhaw/setup-wiremock-action/blob/master/index.js</a></p>

**Comment by Yanjingzhu** [comment, marked as answer]
<p>Hi <a class="mention" href="/williamhaw">@williamhaw</a> , </p>
<p>I checked core.setFailed() in my side, it will mark the step as failed. </p>
<p>Your issue is caused by " <strong>continue-on-error: true</strong>" in your Action E2E Test step , it will mark the step succeed.  </p>
<pre><code>- name: Action E2E Test
        uses: ./
        id: setup-wiremock
        with:
          mappings: 'example-mapping-directory'
          files: 'example-files-directory'
          command: 'node example.test.js'
          verbose: true
        # continue-on-error: true
      - name: Get the WireMock standard output
        if: always()
        run: echo "${{ steps.setup-wiremock.outputs.wiremock-stdout }}"
</code></pre>
<p>You could remove the continue-on-error option. </p>
<p>If you still want to run the next step, you could add if: always() in the next step. </p>

**Comment by williamhaw** [comment]
<p>Aha PEBKAC strikes again! I originally added continue-on-error because I needed to debug something but it’s no longer needed. Thanks!</p>
